src/components/app.tsx

```diff
@@ -168,7 +168,10 @@ export class AppComponent extends BaseComponent<IProps, IState> {
           // use userId as groupId
           db.joinGroup(user.id);
         }
-        else {
+        //only show the group chooser if the logged in user is a student with no group
+        //right now the only difference between the preview URL and the student URL is that the
+        //student url has the `learner` query param, and a token.
+        else if (urlParams === "learner") {
```

Comment:
I hope you're right that it's that simple. It seems like this might adversely affect other non-authenticated modes, e.g. `dev`, `demo`, `qa`, etc. that won't have `learner` parameters but that might need to get to the group chooser dialog. Is it possible to configure (via authoring rather than code) the url parameters passed by the preview button?

Reply:
That may explain the failing tests! I knew it was too easy.

Reply:
I think we may need to detect the preview situation, e.g. no `appMode` and no `token` and force it into a particular mode, either `demo` or `dev`, for instance. and then maybe treat it like `autoAssignStudentsToIndividualGroups` is true.